languages/elixir/exercises/concept/exceptions/.docs/introduction.md

```diff
@@ -0,0 +1,52 @@
+All errors in Elixir implement the _Exception Behaviour_. Just like the _Access Behaviour_, the _Exception Behavior_ defines callback functions that a module must implement to fulfill the software contract of the behaviour. Once an errors is defined, it has the following properties:
+
+- The module's name defines the error's name.
+- The module defines an error-struct.
+- The module will have a `:message` field.
+- The module can be be used with `raise/1` and `raise/2` to raise the intended error
+
+The _Exception Behaviour_ also specifies two required callbacks: `message/1` and `exception/1`. Despite being required, `message/1` is often left to be implemented by the _behavior_, but it is common to implement `exception/1` instead to get custom error messages.
```

Comment:
> but it is common to implement `exception/1` instead to get custom error messages.

I took a look at the standard library https://github.com/elixir-lang/elixir/blob/e8cb3489c91923989d7570ae75fa0c8deae49573/lib/elixir/lib/exception.ex and I am not sure if this statement is true. It looks like it is common to implement `message/1` to get a custom error message, not `exception/1`. But I don't really understand what is the difference between those two callbacks and their intended use.

Reply:
This [comment in the exception source](https://github.com/elixir-lang/elixir/blob/b56c262db3aa32472d244bc2f8ca85df115c5e44/lib/elixir/lib/exception.ex#L676-L679) is an interesting one though that gives it some insight to that discrepancy: 

> ```elixir
> # Some exceptions implement "message/1" instead of "exception/1" mostly
> # for bootstrap reasons. It is recommended for applications to implement
> # "exception/1" instead of "message/1" as described in "defexception/1"
> # docs.
> ```